de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/list/TypeSafeEReferenceArrayInstanceList.java

```diff
@@ -0,0 +1,217 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.concept.list;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.eclipse.emf.ecore.EObject;
+
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyEReference;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.APropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.ArrayInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.EReferencePropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.PropertyinstancesFactory;
+
+public class TypeSafeEReferenceArrayInstanceList<ETYPE extends EObject> extends AArrayInstanceList<BeanPropertyEReference<ETYPE>> {
+
+	protected Class<ETYPE> eClass;
+
+	public TypeSafeEReferenceArrayInstanceList() {
+		super();
+	}
+
+	public TypeSafeEReferenceArrayInstanceList(ArrayInstance ai) {
+		super(ai);
+	}
+	
+	
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean contains(Object o) {
+		if (o instanceof BeanPropertyEReference<?>) {
+			BeanPropertyEReference<ETYPE> categoryBean = (BeanPropertyEReference<ETYPE>) o;
+			APropertyInstance ca = categoryBean.getTypeInstance();
+			for (APropertyInstance pi : ai.getArrayInstances()) {
+				if (pi.getUuid().equals(ca.getUuid())) {

```

Comment:
This has been implemented in the lists to avoid issues in case objects get reloaded in between from the DLVM model.  Than you have different instances with the same UUID. 

The question is, if the assumption back in the days was correct that there can be cases where we have this constellation of semantitcally same objects but as different instances. And we should think about the implications.

For me this is a more general question. And i guess this is taken from the other list implementations. Therefore i would accept it here and take the discussion to some other place.

Reply:
Yep, I simply adopted the way we handle this in the other property instance lists...